vyxal/helpers.py

```diff
@@ -355,6 +355,24 @@ def graph_distance(
         )
 
 
+def group_by_function(
+    lst: VyList, function: types.FunctionType, ctx: Context
+) -> LazyList:
+    """Group a list of elements by a function"""
+
+    def gen():
+        ret = {}
+        for el in lst:
+            key = safe_apply(function, el, ctx=ctx)
+            if key in ret:
+                ret[key].append(el)
+            else:
+                ret[key] = [el]
+        yield from ret.values()
+
+    return gen()
```

Comment:
I meant to take out the `gen` function entirely, you sussy bakas
```suggestion
    ret = {}
    for el in lst:
        key = safe_apply(function, el, ctx=ctx)
        if key in ret:
            ret[key].append(el)
        else:
            ret[key] = [el]
    return ret.values()
```